torch/_dynamo/backends/distributed.py

```diff
@@ -329,27 +313,47 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
         debug_str += "\n---------------\n"
         ddp_graph_log.debug(debug_str)
 
-        # 3: compile each of the partitioned submodules using the user-provided compiler
+        # 3: create a wrapper which lazily compiles each of the partitioned submodules
+        # using the user-provided compiler
         class SubmodCompiler(torch.fx.interpreter.Interpreter):
             def __init__(self, module, compiler):
                 super().__init__(module)
                 self.compiler = compiler
 
-            def compile_submod(self, input_mod, args, kwargs):
+            def lazily_compile_submod(self, input_mod):
                 """
                 Compile the submodule,
                 using a wrapper to make sure its output is always a tuple,
```

Comment:
i think we should rename/cleanup this wrapper class a bit more.

In my other PR (https://github.com/pytorch/pytorch/pull/114186) i renamed it to 'UnwrapperModule' bc it seems like that's more accurate.  And i updated the comments to explain more why i think we need the unwrapping to happen.

In this case maybe just call it a LazyCompileModule or something instead, but still good to take some of my updated comments.

Reply:
Sounds good.